### DISCUSSION TITLE
explicitly typed defaultValues

### DISCUSSION THREAD

**Discussion by melonges** [TanStack/form#609] [Q&A] [answered]
```ts
  const {
    Provider,
    Field,
    state: formState,
  } = useForm({
    defaultValues: {
      commenceTime: null,
    },
  });
```
is defaultValues only being infer? can't it be explicitly typed?

**Comment by crutchcorn** [comment, marked as answer]
~~`useForm<SomeTypeHere>`~~

Cast defaultValues `as` SomeType

**Reply by myeljoud** [reply]
In the latest version, I’m running into this error: `Expected 10 type arguments, but got 1.` Any idea why that’s happening?

I have an input of `type="number"` (amount input), and I don’t want to set a default value of `0`. But since the component is controlled, it complains when I leave it `undefined`. What’s the best way to handle this?

P.S. I’m validating with Zod using `z.number()`.

**Reply by OwenVey** [reply]
Seeing the same issue as well. Trying to find an elegant way to not have to set a default value for number fields and ideally be able to handle `.nullable()` `string` fields

**Reply by crutchcorn** [reply]
Sorry, the method has changed since. Just use `as` to cast defaultValues
